processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -0,0 +1,1321 @@
+    );
+    timeWriter.open();
+    return timeWriter;
+  }
+
+  private ArrayList<GenericColumnSerializer> setupMetricsWriters(
+      final IOPeon ioPeon,
+      final List<String> mergedMetrics,
+      final Map<String, ValueType> metricsValueTypes,
+      final Map<String, String> metricTypeNames,
+      final IndexSpec indexSpec
+  ) throws IOException
+  {
+    ArrayList<GenericColumnSerializer> metWriters = Lists.newArrayListWithCapacity(mergedMetrics.size());
+    final CompressedObjectStrategy.CompressionStrategy metCompression = indexSpec.getMetricCompressionStrategy();
+    for (String metric : mergedMetrics) {
+      ValueType type = metricsValueTypes.get(metric);
+      GenericColumnSerializer writer;
+      switch (type) {
+        case LONG:
+          writer = LongColumnSerializer.create(ioPeon, metric, metCompression);
+          break;
+        case FLOAT:
+          writer = FloatColumnSerializer.create(ioPeon, metric, metCompression);
+          break;
+        case COMPLEX:
+          final String typeName = metricTypeNames.get(metric);
+          ComplexMetricSerde serde = ComplexMetrics.getSerdeForType(typeName);
+          if (serde == null) {
+            throw new ISE("Unknown type[%s]", typeName);
+          }
+          writer = ComplexColumnSerializer.create(ioPeon, metric, serde);
+          break;
+        default:
+          throw new ISE("Unknown type[%s]", type);
+      }
+      writer.open();
+      metWriters.add(writer);
+    }
+    return metWriters;
+  }
+
+  private ArrayList<IndexedIntsWriter> setupDimensionWriters(
+      final IOPeon ioPeon,
+      final List<String> mergedDimensions,
+      final List<ColumnCapabilitiesImpl> dimCapabilities,
+      final Map<String, Integer> dimCardinalities,
+      final IndexSpec indexSpec
+  ) throws IOException
+  {
+    ArrayList<IndexedIntsWriter> dimWriters = Lists.newArrayListWithCapacity(mergedDimensions.size());
+    final CompressedObjectStrategy.CompressionStrategy dimCompression = indexSpec.getDimensionCompressionStrategy();
+    for (int dimIndex = 0; dimIndex < mergedDimensions.size(); ++dimIndex) {
+      String dim = mergedDimensions.get(dimIndex);
+      int cardinality = dimCardinalities.get(dim);
+      ColumnCapabilitiesImpl capabilities = dimCapabilities.get(dimIndex);
+      String filenameBase = String.format("%s.forward_dim", dim);
+      IndexedIntsWriter writer;
+      if (capabilities.hasMultipleValues()) {
+        writer = (dimCompression != null)
+                 ? CompressedVSizeIndexedV3Writer.create(ioPeon, filenameBase, cardinality, dimCompression)
+                 : new VSizeIndexedWriter(ioPeon, filenameBase, cardinality);
+      } else {
+        writer = (dimCompression != null)
+                 ? CompressedVSizeIntsIndexedWriter.create(ioPeon, filenameBase, cardinality, dimCompression)
+                 : new VSizeIndexedIntsWriter(ioPeon, filenameBase, cardinality);
+      }
+      writer.open();
```

Comment:
can we add a comment here about the fact these are closed in another method later on? To make it easier to read for other folks working on this code later on